### PR TITLE
SearchIO query pop: add test and code cleanup

### LINKED ISSUE #3604: SearchIO model pop has untested and potentially unused parameter default
I am not quite sure what to make of the default parameter here:

https://github.com/biopython/biopython/blob/870f623005ed1b95df877f618cbb12a815bd5696/Bio/SearchIO/_model/query.py#L599-L666

I don't see the added value of adding this default. Am I missing something? 

A simple `popped = d.pop() if d else default` does the trick without all the extra object instantiation and testing.

I suggest removing the default parameter to make the pop behave as a pop on a builtin dict.

### PR DESCRIPTION
**Pull request by fabianegli**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #3604

### CODE REVIEW

**Comment by fabianegli** [comment]
I don't get why codecov reports a test coverage decrease when clearly no code was added, but two tests were. Can someone enlighten me?

**Comment by peterjc** [comment]
We see small fluctuations in CodeCov values normally, likely a sign of stochastic branches and/or random test data.

This case could be a side effect of our transition from TravisCI to GitHub actions (were right now I think less dependencies are setup, and thus less code gets tested)

File: Tests/test_SearchIO_model.py [hunk 1]
```diff
@@ -647,6 +647,18 @@ def test_hsp_map_no_func(self):
         self.assertEqual(1102, mapped.seq_len)
         self.assertEqual("refseq_rna", mapped.target)
 
+    def test_pop_inexistent_with_defualt(self):
+        """Test QueryResult.pop with default for inexistent key."""
+        default = "An arbitrary default return value for this test only."
+        inexistent_key = "neither a standard nor alternative key"
+        hit = self.qresult.pop(inexistent_key, default)
```
Comment (bow) [inline Tests/test_SearchIO_model.py:654]:
```suggestion
    def test_pop_nonexistent_with_default(self):
        """Test QueryResult.pop with default for nonexistent key."""
        default = "An arbitrary default return value for this test only."
        nonexistent_key = "neither a standard nor alternative key"
        hit = self.qresult.pop(nonexistent_key, default)
```

File: Tests/test_SearchIO_model.py [hunk 2]
```diff
@@ -647,6 +647,18 @@ def test_hsp_map_no_func(self):
...
+        self.assertEqual(hit, default)
+
+    def test_pop_inexistent_key(self):
+        """Test QueryResult.pop with default for inexistent key."""
+        inexistent_key = "neither a standard nor alternative key"
+        self.assertRaises(KeyError, self.qresult.pop, inexistent_key)
```
Comment (bow) [inline Tests/test_SearchIO_model.py:660]:
```suggestion
    def test_pop_nonexistent_key(self):
        """Test QueryResult.pop with default for nonexistent key."""
        nonexistent_key = "neither a standard nor alternative key"
        self.assertRaises(KeyError, self.qresult.pop, nonexistent_key)
```

**Review by bow — CHANGES_REQUESTED** [review]
@fabianegli Thanks for opening this MR :+1: ~ I only have minor comments related to spelling, back to you.

**Comment by fabianegli** [comment]
@bow Thanks for the review and all your input!

**Comment by bow** [comment]
@fabianegli Thanks for pushing the changes ~ looks good to me now and have added my approval :).

@peterjc I can see test failures but they seem to be unrelated to the changes made in this MR. Are there any changes from `master` that should be merged back in, or are those failures intermittent?

**Comment by peterjc** [comment]
There are some known glitches related to adopting GitHub actions (or just coincidental in timing) like #3640 and #3659

**Comment by fabianegli** [comment]
Now that the test coverage calculation issue has been resolved in the repo this is probably ready to be merged?

**Comment by bow** [comment]
@fabianegli I believe so :+1: ~ let's go ahead with this.

**Pull request merged**